ontotext-yasgui-web-component/src/components/ontotext-tooltip-web-component/ontotext-tooltip-web-component.tsx

```diff
@@ -9,27 +9,115 @@ const TOOLTIP_CLASS_NAME = 'ontotext-yasgui-tooltip';
   shadow: false,
 })
 export class OntotextTooltipWebComponent {
-
   @Element() el: HTMLElement;
 
-  @Prop() dataTooltip: string;
-
+  @Prop() yasguiDataTooltip: string;
   @Prop() placement: string;
-
   @Prop() showOnClick: false;
 
-  @Watch('dataTooltip')
+  @Watch('yasguiDataTooltip')
   configurationChanged() {
-    this.update();
+    // Only update content if tooltip exists
+    if (this.boxEl) {
+      this.boxEl.innerHTML = this.yasguiDataTooltip || '';
+    }
   }
 
-  private tooltip: Instance;
+  private popperEl: HTMLElement | null = null;
+  private boxEl: HTMLElement | null = null;
+  private arrowEl: HTMLElement | null = null;
+  private cleanupAutoUpdate: (() => void) | null = null;
+
+  private onMouseOver: ((e: MouseEvent) => void) | null = null;
+  private onMouseLeave: ((e: MouseEvent) => void) | null = null;
+  private onDocClick: ((e: MouseEvent) => void) | null = null;
+
+  private unsubscriptions: Array<() => void> = [];
+
+  private show = () => {
+    if (!this.yasguiDataTooltip) {
+      return;
+    }
+    this.ensurePopperCreated();
+    if (!this.popperEl) {
+      return;
+    }
+
+    this.popperEl.style.visibility = 'visible';
+    this.popperEl.style.pointerEvents = 'none';
+
+    this.updatePosition();
+    this.startAutoUpdate();
+
+    if (this.showOnClick && !this.onDocClick) {
+      const handler: (e: MouseEvent) => void = (e) => {
+        const target = e.target as Node | null;
+        const insideHost = !!(target && this.el.contains(target));
+        const insidePopper = !!(target && this.popperEl && this.popperEl.contains(target));
+        if (!insideHost && !insidePopper) {
+          this.hide();
+        }
+      };
+      this.onDocClick = handler;
+      document.addEventListener('click', handler, true);
+
+      this.unsubscriptions.push(() => {
+        document.removeEventListener('click', handler, true);
+        if (this.onDocClick === handler) this.onDocClick = null;
+      });
+    }
+  };
+
+  private hide = () => {
+    if (!this.popperEl) {
+      return;
+    }
 
-  private showFunction;
-  private hideFunction;
+    this.stopAutoUpdate();
+
+    if (this.popperEl.parentNode) {
+      this.popperEl.parentNode.removeChild(this.popperEl);
+    }
+
+    this.popperEl = null;
+    this.boxEl = null;
+    this.arrowEl = null;
+  };
 
   componentDidLoad() {
-    this.init();
+    this.onMouseOver = (e: MouseEvent) => {
+      const from = e.relatedTarget as Node | null;
+      if (!from || !this.el.contains(from)) {
+        this.show();
+      }
+    };
+    this.onMouseLeave = (e: MouseEvent) => {
+      const to = e.relatedTarget as Node | null;
+      if (!to || !this.el.contains(to)) {
+        this.hide();
+      }
+    };
+
+    if (this.onMouseOver) {
+      this.el.addEventListener('mouseover', this.onMouseOver);
+    }
+    this.unsubscriptions.push(() => {
```

Comment:
This is a little strange (the same applies to the next one). The event listener is added if this.onMouseOver exists, but the unsubscription always. Move it inside the if. In my opinion, this if and the next one are redundant: componentDidLoad starts with the creation of onMouseOver and onMouseLeave, so they are always defined.